### PR TITLE
Add configurable-http-proxy

### PR DESCRIPTION
**Pull request by minrk**
A nodejs package!

### CODE REVIEW

File: recipes/configurable-http-proxy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  name: configurable-http-proxy
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/jupyterhub/configurable-http-proxy.git
+  git_tag: 57fe19e0700c8bd3af0f77d8b7f51a54aa02fc87
+
+build:
+  number: 0
+  script: npm install -g .
```
Comment (pelson) [inline recipes/configurable-http-proxy/meta.yaml]: It is a brave new world... and so conda-forge begins packaging node packages 🎉 

Out of interest, have you had a look at the directory structure, and where the global directory actually is with this build? (I will take a look myself in the morning, but I'm just curious)

File: recipes/configurable-http-proxy/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{% set version = "1.2.0" %}
+
+package:
+  name: configurable-http-proxy
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/jupyterhub/configurable-http-proxy.git
```
Comment (pelson) [inline recipes/configurable-http-proxy/meta.yaml]: I guess we could just take the tarball from github, right?

Comment (pelson) [inline recipes/configurable-http-proxy/meta.yaml]: https://github.com/jupyterhub/configurable-http-proxy/archive/1.2.0.tar.gz

**Comment by pelson** [comment]
This is looking good, and is the first node package on conda-forge. 🎉

**Comment by minrk** [comment]
Updated to use github archive tarball.

**Comment by ocefpaf** [comment]
LGTM. Thanks @minrk!

**Comment by pelson** [comment]
I was tempted to add a ```node-``` prefix to the name so that we can start namespacing the various sources of packages. I don't think we need to do that immediately, but if we get lots of node packages, it may be something we should consider.

**Comment by ocefpaf** [comment]
> I was tempted to add a ```node-``` prefix to the name so that we can start namespacing the various sources of packages. I don't think we need to do that immediately, but if we get lots of node packages, it may be something we should consider.

Sorry. I was to hasty to merge.

I agree about addng the name space, but I disagree that we don't have to do it immediately. The more we wait the harder it will be to do it later on. If @minrk is OK with it I can rename the feedstock repository and send a PR to fix this. (Or maybe delete the feedstock and re-create submit it to staged-recipes.)

**Comment by msarahan** [comment]
Where do namespaces start and end? Should python stuff be namespaced, too?  IMHO that depends on whether we're a python package collection that supports other languages to some extent (true right now), or we're a general purpose cross platform collection of packages (going here?)

We have discussed this at continuum, but have not settled on anything.

**Comment by ocefpaf** [comment]
@msarahan I added a topic about this for our next meeting. I don't think this is an easy thing to do. Just look how different the Linuxes distros solution are. However, there are some low hanging fruits. For example: we don't have many node packages, so adding the namespace to node packages right now is a good idea.

Python namespaces are be a headache due to many reasons. Here are a few I can think of:

- user familiarity with the project name. (OpenSUSE users are still confused that they must install `python-numpy` or `python3-numpy` instead of just `numpy`.)
- `conda-forge` should not deviate (too much) from the names used in the default channel.
- should we do it for all packages or just those that wrapped a collection of languages?
- should the prefix be `python-`? `py-`?

**Comment by pelson** [comment]
> Sorry. I was to hasty to merge.

Not at all. We shouldn't be precious about merging sensible proposals and should always be able to iterate after the fact.

> Where do namespaces start and end?

I previously jokingly proposed the following: ```if 'py' not in name: name = 'python-{}'.format(name)```

It actually does hit the 80-20 rule, and might be something we consider (to be discussed on Friday).

> If @minrk is OK with it I can rename the feedstock repository

OK, let's get @minrk's feedback, and then my preference it to resubmit a recipe (we need to look in more detail at how easy it is to rename feedstocks, and I haven't yet had a chance to do so).

**Comment by ocefpaf** [comment]
> I previously jokingly proposed the following: ```if 'py' not in name: name = 'python-{}'.format(name)```

I know :smile:, but we must lay out the problem and discuss it anyways to be prepared for the future. Even if we never adopt the `python-` namespace.

**Comment by minrk** [comment]
My preference (not a strong one) would be to not prefix, mainly because the primary goal of the package is to provide the `configurable-http-proxy` executable, rather than a js library. A related example would be `cython` on debian, which is not `python-cython`, but `cython`, whereas `python-numpy` provides a library.

At the same time, if all (non-Python) packages are `language-` prefixed, that might be simpler for everyone (I know I've tried to `apt-get install python3-cython` before). If we do that, it seems a bit wrong that Python packages are not prefixed, but conda really is Python-centric, even though it *supports* packages in other languages.

**Comment by pelson** [comment]
> mainly because the primary goal of the package is to provide the configurable-http-proxy executable

This is a fair point. There is no point in us namespacing the package if the executable itself isn't namespaced - it isn't like we can have another package which provides a ```configurable-http-proxy``` in the bin directory.

Let's hold fire on renaming. I suspect there *will* be cases where this is much more clear cut, at which point we should reassess.

**Pull request merged**